test/src/dftbp/api/mm/testers/CMakeLists.txt

```diff
@@ -14,6 +14,17 @@ if(WITH_C_EXECUTABLES)
   add_executable(test_fileinitc test_fileinitc.c $<TARGET_OBJECTS:test_api_mm_utils>)
   target_link_libraries(test_fileinitc dftbplus)
   set_target_properties(test_fileinitc PROPERTIES LINKER_LANGUAGE C)
+
+  add_executable(test_expdmhs test_expdmhs.c $<TARGET_OBJECTS:test_api_mm_utils>)
+  target_link_libraries(test_expdmhs dftbplus)
+  set_target_properties(test_expdmhs PROPERTIES LINKER_LANGUAGE C)
+  
+  if(WITH_MPI)
+    find_package(MPI REQUIRED)
+    add_executable(test_mpiexpdmhs test_mpiexpdmhs.c $<TARGET_OBJECTS:test_api_mm_utils>)
+    target_link_libraries(test_mpiexpdmhs dftbplus)
```

Comment:
`
target_link_libraries(test_mpiexpdmhs dftbplus ${MPI_C_LIBRARIES})
`
Possibly

Reply:
I think, it should be rather
```
target_link_libraries(... MPI::MPI_C)
```
As the target `dftbplus` only depends on `MPI::MPI_Fortran`, the C-libraries are probably indeed missing here.

@PavelStishenko Yes, if you set `CC=mpicc`, then it works out of the box. However, it should also work with `CC=gcc` and then CMake (hopefully) finding the right MPI-dependencies.

Reply:
`tests_mpiexpdmhs.c` pass on my machine. 
I use this command for cmake config:
`LDFLAGS="-Wl,--copy-dt-needed-entries" FC=gfortran-8 CC=gcc-8 FFLAGS="-w" cmake -DLAPACK_LIBRARY=openblas -DSCALAPACK_LIBRARY=scalapack-openmpi -DCMAKE_INSTALL_PREFIX=${HOME}/opt/dftbp-mpi2 -B _build-mpi2 -C mpi.config.cmake .`

And this config `mpi.config.cmake`: https://pastebin.com/9rELuLU5